### PR TITLE
Add support of NonEmptySet into the cat's module.

### PR DESCRIPTION
**Pull request by NicolaeNMV**
Cats 1.1.0 added the support of `NonEmptySet`. Add the pureconfig convertors to support it.

### CODE REVIEW

**Review by ruippeixotog — CHANGES_REQUESTED** [review]
Thanks for your PR, @NicolaeNMV, looks good!

Can you just add tests for the `NonEmptySet` writer you wrote, so that Coveralls doesn't complain? It would also be awesome if you added writer tests for the other non-empty data structures :)

**Comment by NicolaeNMV** [comment]
@ruippeixotog thanks for reviewing this so quickly ;)

I've added the missing tests plus did some refactoring to make the tests easier to read.

**Review by ruippeixotog — APPROVED** [review]
LGTM! Thanks again!

File: modules/cats/src/main/scala/pureconfig/module/cats/package.scala [hunk 1]
```diff
@@ -31,4 +32,10 @@ package object cats {
 
   implicit def nonEmptyVectorWriter[T](implicit vectorWriter: ConfigWriter[Vector[T]]): ConfigWriter[NonEmptyVector[T]] =
     ConfigWriter.fromFunction(nonEmptyVector => vectorWriter.to(nonEmptyVector.toVector))
+
+  implicit def nonEmptySetReader[T](implicit listReader: ConfigReader[SortedSet[T]]): ConfigReader[NonEmptySet[T]] =
```
Comment (leifwickland) [inline modules/cats/src/main/scala/pureconfig/module/cats/package.scala]: Nitpick: `listReader` is slightly misnamed. It's probably better called a `setReader`. Ditto on 39.

**Review by leifwickland — APPROVED** [review]
Thanks for this nice addition, @NicolaeNMV, and for improving the test coverage.

I sent along a trivial suggestion.

Approved either way.

**Comment by NicolaeNMV** [comment]
@leifwickland I've slightly changed the naming of the vals to avoid repetition of the information that is in the type. I also made a better use of new lines to group the reader and the writer together. Can revert if that's too much. WDYT ?

**Comment by leifwickland** [comment]
@NicolaeNMV That's even better than what I suggested. Thank you.

**Comment by leifwickland** [comment]
@NicolaeNMV Thanks again for the contribution. Would you like to send a PR to add yourself to the [pureconfig contributors list](https://github.com/pureconfig/pureconfig/blob/6b6e0ffb118f1b80fa2254c0ddc4679cdfdc4837/CONTRIBUTORS)?
